src/middlewared/middlewared/etc_files/nfsd.py

```diff
@@ -5,13 +5,13 @@
 logger = logging.getLogger(__name__)
 
 
-async def get_exports(config, shares, kerberos_keytabs):
+async def get_exports(config, shares, has_nfs_principal):
     result = []
 
     if config["v4"]:
         if config["v4_krb"]:
             result.append("V4: / -sec=krb5:krb5i:krb5p")
-        elif kerberos_keytabs:
+        elif has_nfs_principal:
```

Comment:
So, in theory, if the user doesn't have kerberos authentication enabled but they have NFS SPN entries, then we generate /etc/exports so. We're still starting gssd with either situation right? Because if the user
1. doesn't enable kerberos auth
2. but has an NFS SPN entry
Do we start gssd?

Reply:
That's a good question. Underlying logic isn't substantively changed except to be aware that not all keytabs are for NFS.

Prior to this PR though an AD user with v4 enabled _always_ had "V4: / -sec=sys:krb5:krb5i:krb5p" even when no krb5 was intended for shares.